packages/eslint-plugin/src/rules/no-duplicate-type-constituents.ts

```diff
@@ -117,115 +119,158 @@ export default createRule<Options, MessageIds>({
     const parserServices = getParserServices(context);
     const { sourceCode } = context;
 
+    function report(
+      messageId: MessageIds,
+      constituentNode: TSESTree.TypeNode,
+      data?: Record<string, unknown>,
+    ): void {
+      const getUnionOrIntersectionToken = (
+        where: 'After' | 'Before',
+        at: number,
+      ): TSESTree.Token | undefined =>
+        sourceCode[`getTokens${where}`](constituentNode, {
+          filter: token =>
+            ['&', '|'].includes(token.value) &&
+            constituentNode.parent.range[0] <= token.range[0] &&
+            token.range[1] <= constituentNode.parent.range[1],
+        }).at(at);
+
+      const beforeUnionOrIntersectionToken = getUnionOrIntersectionToken(
+        'Before',
+        -1,
+      );
+      let afterUnionOrIntersectionToken: TSESTree.Token | undefined;
+      let bracketBeforeTokens;
+      let bracketAfterTokens;
+      if (beforeUnionOrIntersectionToken) {
+        bracketBeforeTokens = sourceCode.getTokensBetween(
+          beforeUnionOrIntersectionToken,
+          constituentNode,
+        );
+        bracketAfterTokens = sourceCode.getTokensAfter(constituentNode, {
+          count: bracketBeforeTokens.length,
+        });
+      } else {
+        afterUnionOrIntersectionToken = nullThrows(
+          getUnionOrIntersectionToken('After', 0),
+          NullThrowsReasons.MissingToken(
+            'union or intersection token',
+            'duplicate type constituent',
+          ),
+        );
+        bracketAfterTokens = sourceCode.getTokensBetween(
+          constituentNode,
+          afterUnionOrIntersectionToken,
+        );
+        bracketBeforeTokens = sourceCode.getTokensBefore(constituentNode, {
+          count: bracketAfterTokens.length,
+        });
+      }
+      context.report({
+        loc: {
+          start: constituentNode.loc.start,
+          end: (bracketAfterTokens.at(-1) ?? constituentNode).loc.end,
+        },
+        node: constituentNode,
+        messageId,
+        data,
+        fix: fixer =>
+          [
+            beforeUnionOrIntersectionToken,
+            ...bracketBeforeTokens,
+            constituentNode,
+            ...bracketAfterTokens,
+            afterUnionOrIntersectionToken,
+          ].flatMap(token => (token ? fixer.remove(token) : [])),
+      });
+    }
+
+    function checkDuplicateRecursively(
+      unionOrIntersection: UnionOrIntersection,
+      constituentNode: TSESTree.TypeNode,
+      uniqueConstituents: TSESTree.TypeNode[],
+      cachedTypeMap: Map<Type, TSESTree.TypeNode>,
+      forEachNodeType?: (type: Type, node: TSESTree.TypeNode) => void,
+    ): void {
+      const type = parserServices.getTypeAtLocation(constituentNode);
+      if (tsutils.isIntrinsicErrorType(type)) {
+        return;
+      }
+      const duplicatedPrev =
```

Comment:
[Naming] Abbreviations make the code a little harder to understand, especially for folks who don't have the same intuition for what they mean. Let's stick with full words. Clarity at the cost of verbosity.

```suggestion
      const duplicatedPrevious =
```